sklearn/preprocessing/_csr_hstack.pyx

```diff
@@ -0,0 +1,103 @@
+# distutils: language=c++
+# Based on `scipy.sparse.sparsetools.csr.h` under the BSD license.
+
+cimport cython
+from libc.stdlib cimport malloc, free
+
+cimport numpy as cnp
+cnp.import_array()
+
+# Fused type for data
+ctypedef fused DATA_t:
+    cython.numeric
```

Comment:
This includes complex numbers as well. For our use case, I think we only need support for `float{32,64}`, `int{32,64}`